book/src/concepts/system.md

```diff
@@ -290,6 +290,179 @@ write_storage.remove(entity);
 
 Keep in mind that inserting a component on an entity that already has a component of the same type **will overwrite the previous one**.
 
+## Changing states through resources
+
+In a previous section we talked about [`States`][s], and how they are used to organize your game
+into different logical sections.
+Sometimes we want to trigger a state transition from a system.
+For example, if a player dies we might want to remove their entity and signal to the state machine
+to push a state that shows a "You Died" screen.
+
+So how can we affect states from systems?
+There are a couple of ways, but this section will detail the easiest one: using a [`Resource`][r].
+
+Before that, let's just quickly remind ourselves what a resource is:
+
+> A [`Resource`][r] is any type that stores data that you might need for your game AND that is not
+> specific to an entity.
+
+The data in a resource is available both to systems and resources.
+We can use this to our advantage!
+
+Let's say you have the following two states:
+
+* `GameplayState`: State in which the game is running.
+* `GameMenuState`: State where the game is paused and we interact with a game menu.
+
+The following example shows how to keep track of which state we are currently in.
+This allows us to do a bit of conditional logic in our systems to determine what to do depending on
+which state is currently active. And manipulating the states by setting flags:
```

Comment:
", and"

Reply:
Sentence would become a bit long. Do you think this is an issue?

Reply:
Nah I think it will be fine

Reply:
Ok, done!